terraform/lang/eval.go

```diff
@@ -256,14 +140,6 @@ func (s *Scope) evalContext(refs []*addrs.Reference, selfAddr addrs.Referenceabl
 		return ctx, diags
 	}
 
-	// First we'll do static validation of the references. This catches things
-	// early that might otherwise not get caught due to unknown values being
-	// present in the scope during planning.
-	if staticDiags := s.Data.StaticValidateReferences(refs, selfAddr); staticDiags.HasErrors() {
```

Comment:
`StaticValidateReferences` mainly detect reference errors of resources, data attributes, module attributes early. However, TFLint does not support the evaluation of these attributes.